IntentsExtension/IntentHandlers/AppendNoteIntentHandler.swift

```diff
@@ -27,14 +27,24 @@ class AppendNoteIntentHandler: NSObject, AppendNoteIntentHandling {
             return AppendNoteIntentResponse(code: .failure, userActivity: nil)
         }
 
-        guard let existingContent = try? await Downloader(simperiumToken: token).getNoteContent(for: identifier) else {
-            return AppendNoteIntentResponse(code: .failure, userActivity: nil)
+        do {
+            let existingContent = try await Downloader(simperiumToken: token).getNoteContent(for: identifier)
+            note.content = existingContent + "\n\(content)"
+        } catch {
+            return handleFailure(with: error, content: content)
         }
 
-        note.content = existingContent + "\n\(content)"
         let uploader = Uploader(simperiumToken: token)
-        uploader.send(note)
+        do {
+            _ = try await uploader.send(note)
+            return AppendNoteIntentResponse(code: .success, userActivity: nil)
+        } catch {
+            return handleFailure(with: error, content: content)
+        }
+    }
```

Comment:
There's a bit of duplication in this API. You don't **really** need two Try / Catch (since if any of them both fails, you'll bail out.

Alternative A is something like this:

```
        // ...

        do {
            let downloader = Downloader(simperiumToken: token)
            let existingContent = try await downloader.getNoteContent(for: identifier)
            note.content = existingContent + "\n\(content)"

            let uploader = Uploader(simperiumToken: token)
            try await uploader.send(note)
        } catch {
            return handleFailure(with: error, content: content)
        }

        return AppendNoteIntentResponse(code: .success, userActivity: nil)
    }
```

Idea B is to just extract the `download` + `upload` API(s) into smaller methods, and it would look like...

```
        // ...

        do {
            let existingContent = try await downloadNoteContent(for: identifier, simperiumToken: token)
            note.content = existingContent + "\n\(content)"

            try await uploadNoteContent(note, simperiunToken: token)
        } catch {
            return handleFailure(with: error, content: content)
        }

        return AppendNoteIntentResponse(code: .success, userActivity: nil)
    }
```